### PR TITLE
updated languages and territories

### PR DESCRIPTION
**Pull request by fritzschoff**

### CODE REVIEW

File: libs/movie/src/lib/movie/form/movie.form.ts [hunk 1]
```diff
@@ -44,11 +44,10 @@ import {
   createHostedVideos,
   createHostedVideo
 } from '../+state/movie.model';
-
 import { FormArray, FormControl, Validators, ValidatorFn } from '@angular/forms';
 import { Filmography, createFilmography } from '@blockframes/utils/common-interfaces/identity';
 import { LegalDocument } from '@blockframes/contract/contract/+state/contract.firestore';
-import { FormStaticValue, FormConstantValue } from '@blockframes/utils/form/forms/static-value.form';
+import { FormConstantValue } from '@blockframes/utils/form/forms/static-value.form';
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/movie.form.ts:50]: Hmmm I liked the `Static` name. I think it makes more sense than "constant" which is already something in javascript

File: libs/ui/src/lib/form/language/language.component.ts [hunk 2]
```diff
@@ -17,14 +15,12 @@ export class FormLanguageComponent implements OnInit {
...
-  public languages = languages;
-
-  filteredLanguages$: Observable<LANGUAGES>;
+  filteredLanguages$: Observable<Language[]>;
 
   ngOnInit() {
     this.filteredLanguages$ = this.form.valueChanges.pipe(
-      startWith(''),
-      map(language => (language ? this.filter(language) : this.languages))
+      startWith('' as Language),
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/form/language/language.component.ts:22]: Could you remove the startWith ? Or start with undefined

File: libs/ui/src/lib/form/language/language.component.ts [hunk 3]
```diff
@@ -17,14 +15,12 @@ export class FormLanguageComponent implements OnInit {
...
+      map(language => (language ? this.filter(language) : languages as any))
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/form/language/language.component.ts:23]: Why do you need to strong type languages ? Isn't it a `Language[]` ?

File: libs/utils/src/lib/pipes/to-label.pipe.ts [hunk 4]
```diff
@@ -1,36 +1,25 @@
 import { NgModule, Pipe, PipeTransform } from '@angular/core';
 import { CommonModule } from '@angular/common';
 import { staticConsts } from '@blockframes/utils/static-model';
-import { getValueByKey, Scope } from '../static-model/staticConsts';
-
-export const formatKey = (key: string, scope: Scope) => key
-  ? getValueByKey(scope, key.trim().toLocaleLowerCase())
-  : '';
+import { Scope } from '../static-model/staticConsts';
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/pipes/to-label.pipe.ts:4]: I just realisd the name of the file should be kebab-case

File: libs/utils/src/lib/pipes/to-label.pipe.ts [hunk 5]
```diff
@@ -1,36 +1,25 @@
...
-        if (Array.isArray(value)) {
-          return value.map((v) => formatKey(v, key as Scope)).join(', ');
-        }
-        else return staticConsts[key][value];
-      }
+  transform(value: string | string[], scope: Scope): string | string[] {
+    try {
+      return Array.isArray(value) ? value.map(val => staticConsts[scope][val]).join(', ') : staticConsts[scope][value];
+    } catch (error) {
+      console.error(`Could not find label for ${value}`)
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/pipes/to-label.pipe.ts:14]:
```
console.error(`Could not find label for key "${value}" in scrope "${scope}"`)
```

**Review by GrandSchtroumpf — CHANGES_REQUESTED** [review]

File: libs/ui/src/lib/form/language/language.component.ts (hunk 3 above)
Comment (fritzschoff) [reply libs/ui/src/lib/form/language/language.component.ts:23]: nope, langauges is just the object and langauge is just the keys

File: libs/admin/src/lib/admin-panel/components/edit-title/edit-title.component.html [hunk 6]
```diff
@@ -17,7 +17,7 @@ <h1>{{ data.title }}</h1>
         Amount is <strong>required</strong>
       </mat-error>
     </mat-form-field>
-    <static-select scope="movieCurrencies" type="constant" [withoutValues]="['']" [control]="form.get('currency')">
+    <static-select scope="movieCurrencies" [withoutValues]="['']" [control]="form.get('currency')">
```
Comment (GrandSchtroumpf) [inline libs/admin/src/lib/admin-panel/components/edit-title/edit-title.component.html:20]: Do you need `withoutValues` here ?

File: libs/contract/src/lib/version/form/price/price.component.html [hunk 7]
```diff
@@ -1,6 +1,6 @@
 <!-- Currency -->
 <h5>CONTRACT CURRENCY</h5>
-<static-select scope="movieCurrencies" type="constant" [control]="currencyCtrl" [withoutValues]="['']">
+<static-select scope="movieCurrencies" [control]="currencyCtrl" [withoutValues]="['']">
```
Comment (GrandSchtroumpf) [inline libs/contract/src/lib/version/form/price/price.component.html:3]: same as above

File: libs/ui/src/lib/form/language/language.component.ts (hunk 3 above)
Comment (GrandSchtroumpf) [reply libs/ui/src/lib/form/language/language.component.ts:23]: Hmmm then you've got a type error no ?
If filteredLanguages is supposed to have an array, and if "languages" is an object, you should never return the languages but `Object.keys(languages)` no ?

File: libs/ui/src/lib/static-autocomplete/check-boxes/check-boxes.component.ts [hunk 8]
```diff
@@ -34,17 +32,13 @@ export class StaticCheckBoxesComponent implements OnInit {
   @Output() added = new EventEmitter<string>();
   @Output() removed = new EventEmitter<number>();
 
-  public items: SlugAndLabel[];
+  public items: any;
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/static-autocomplete/check-boxes/check-boxes.component.ts:35]: shouldn't it be at least a string[] ?

File: libs/ui/src/lib/static-autocomplete/input/input-autocomplete.component.ts [hunk 9]
```diff
@@ -17,18 +16,18 @@ import { startWith, map } from 'rxjs/operators';
...
   @Input() mode: 'legacy' | 'standard' | 'fill' | 'outline' = 'outline'
 
-  public _scope: SlugAndLabel[];
+  public state: Scope[];
+
+  public _scope: Scope;
 
-  @Input() set scope(value: string) {
-    this._scope = staticModels[value];
+  @Input() set scope(value: Scope) {
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/static-autocomplete/input/input-autocomplete.component.ts:25]: do a normal @Input() now that we don't do any operation.

**Review by GrandSchtroumpf — APPROVED** [review]

**Pull request merged**